cohortextractor2/validate_dummy_data.py

```diff
@@ -102,9 +102,18 @@ def date_validator(value):
         except ValueError:
             datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
 
-    def category_validator(value, categories):
-        category_type = type(list(categories)[0])
-        if category_type(value) not in categories:
+    def category_validator(value, categories, default_category):
+        """Ensure that a category value is one of the expected categories, or the default"""
+        # The default can be a different type (e.g. None, or the string "missing")
```

Comment:
Isn't the _only_ case where the default should be able to differ in type from the other values the case when it's None (i.e. it becomes SQL NULL)? I don't think we want to allow e.g. integer categories and a default of `"missing"`, do we?

Reply:
We currently do allow it, but we probably shouldn't.  I'll make it validate that here, and add the validation of the default to my other PR